pom.xml

```diff
@@ -1097,6 +1094,13 @@
                               <arg>-XDcompilePolicy=simple</arg>
                               <arg>-Xplugin:ErrorProne -Xep:ThreadLocalUsage:ERROR -Xep:MissingOverride:ERROR -Xep:NonAtomicVolatileUpdate:ERROR -Xep:SynchronizeOnNonFinalField:ERROR -Xep:StaticQualifiedUsingExpression:ERROR -Xep:WaitNotInLoop:ERROR -Xep:BanJNDI:OFF -XepExcludedPaths:.*/generated-sources/.*</arg>
                           </compilerArgs>
+                          <annotationProcessorPaths>
+                              <path>
+                                  <groupId>com.google.errorprone</groupId>
+                                  <artifactId>error_prone_core</artifactId>
+                                  <version>${errorprone.version}</version>
+                              </path>
+                          </annotationProcessorPaths>
```

Comment:
I know that Emmanuel deliberately avoided doing this when overhauling the ErrorProne usage a while back for JDK > 9, as it restricts use of annotation processors from the classpath and requires every processor be specified here in the XML. We do have another annotation processor being used via the classpath (artemis-log-annotation-processor).

I would guess that is what caused the 'full test run' CI failures @clebertsuconic  noted since I know that build enables ErrorProne during compile, meaning it probably then failed to run the other annotation processor as it isnt also specified here (or rather outwith these profiles), and so failed to generate the logger classes and then tripped over their absence.

Since ErrorProne is already isolated to a profile I think I would leave this as it was. At least I dont think it really belongs in the same commit as a swap of caching to use Caffeine.

Reply:
ErrorProne configuration changed because produce `java.lang.NoClassDefFoundError: com/google/common/collect/ImmutableMap` during compilation

Reply:
https://github.com/apache/activemq-artemis/actions/runs/5487090252/jobs/9998059886?pr=4540

Reply:
```
activemq-artemis\artemis-quorum-ri>mvn dependency:tree -Derrorprone

[INFO] --- dependency:3.6.0:tree (default-cli) @ artemis-quorum-ri ---
[INFO] org.apache.activemq:artemis-quorum-ri:jar:2.30.0-SNAPSHOT
[INFO] +- org.apache.curator:curator-recipes:jar:5.2.0:compile
[INFO] |  \- org.apache.curator:curator-framework:jar:5.2.0:compile
[INFO] +- org.apache.curator:curator-client:jar:5.2.0:compile
[INFO] |  \- com.google.guava:guava:jar:27.0.1-jre:compile
[INFO] |     +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |     +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |     +- org.checkerframework:checker-qual:jar:2.5.2:compile
[INFO] |     +- com.google.j2objc:j2objc-annotations:jar:1.1:compile
[INFO] |     \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.17:compile
[INFO] +- org.apache.zookeeper:zookeeper:jar:3.6.3:compile
[INFO] |  +- org.apache.zookeeper:zookeeper-jute:jar:3.6.3:compile
[INFO] |  \- io.netty:netty-handler:jar:4.1.94.Final:compile
[INFO] |     +- io.netty:netty-common:jar:4.1.94.Final:compile
[INFO] |     +- io.netty:netty-resolver:jar:4.1.94.Final:compile
[INFO] |     +- io.netty:netty-transport-native-unix-common:jar:4.1.94.Final:compile
[INFO] |     \- io.netty:netty-codec:jar:4.1.94.Final:compile
[INFO] +- org.apache.curator:curator-test:jar:5.2.0:test
[INFO] |  +- io.dropwizard.metrics:metrics-core:jar:3.2.5:test
[INFO] |  \- org.xerial.snappy:snappy-java:jar:1.1.7:test
[INFO] +- org.apache.activemq:artemis-quorum-api:jar:2.30.0-SNAPSHOT:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.36:compile
[INFO] +- org.apache.logging.log4j:log4j-slf4j-impl:jar:2.20.0:test
[INFO] |  +- org.apache.logging.log4j:log4j-api:jar:2.20.0:test
[INFO] |  \- org.apache.logging.log4j:log4j-core:jar:2.20.0:test
[INFO] +- org.apache.activemq:artemis-commons:jar:2.30.0-SNAPSHOT:compile
[INFO] |  +- io.netty:netty-buffer:jar:4.1.94.Final:compile
[INFO] |  +- io.netty:netty-transport:jar:4.1.94.Final:compile
[INFO] |  \- commons-beanutils:commons-beanutils:jar:1.9.4:compile
[INFO] |     +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |     \- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] +- junit:junit:jar:4.13.2:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- org.apache.activemq:artemis-unit-test-support:jar:2.30.0-SNAPSHOT:test
[INFO] +- org.hamcrest:hamcrest:jar:2.1:test
[INFO] \- com.google.errorprone:error_prone_core:jar:2.20.0:provided
[INFO]    +- com.google.errorprone:error_prone_annotation:jar:2.20.0:provided
[INFO]    +- com.google.errorprone:error_prone_type_annotations:jar:2.20.0:provided
[INFO]    +- com.google.errorprone:error_prone_check_api:jar:2.20.0:provided
[INFO]    |  +- io.github.java-diff-utils:java-diff-utils:jar:4.0:provided
[INFO]    |  |  \- org.eclipse.jgit:org.eclipse.jgit:jar:4.4.1.201607150455-r:provided
[INFO]    |  +- com.github.kevinstern:software-and-algorithms:jar:1.0:provided
[INFO]    |  +- com.github.ben-manes.caffeine:caffeine:jar:3.1.6:provided
[INFO]    |  \- com.google.inject:guice:jar:5.1.0:provided
[INFO]    |     \- aopalliance:aopalliance:jar:1.0:provided
[INFO]    +- org.pcollections:pcollections:jar:3.1.4:provided
[INFO]    +- com.google.auto:auto-common:jar:1.2.1:provided
[INFO]    +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO]    +- io.github.eisop:dataflow-errorprone:jar:3.34.0-eisop1:provided
[INFO]    +- com.google.auto.value:auto-value-annotations:jar:1.9:provided
[INFO]    +- com.google.errorprone:error_prone_annotations:jar:2.20.0:compile
[INFO]    +- com.google.protobuf:protobuf-java:jar:3.19.6:provided
[INFO]    +- com.google.auto.service:auto-service-annotations:jar:1.0.1:provided
[INFO]    \- javax.inject:javax.inject:jar:1:provided
```

Reply:
So that module is now using Guava 27, which presumably doesn't have the thing needed by ErrorProne, which normally uses a newer version.

Maven is picking the older version during resolution, probably because it is earlier on the dependency set than the ErrorProne dep is (it will be last due to being in a profile). Its doing that instead of just using the specified dependencyManagement guava version that was removed from the build. Restoring that may fix things. Alternative perhaps there is a newer curator-client version that uses a recent guava.

Reply:
still I can't reproduce error using `annotationProcessorPaths` approach

Reply:
I'd guess because it is doing a completely different thing there, with the ErrorProne deps defined earlier off-classpath, likely meaning its transitive guava dep version gets picked up, before starting to look at the compile deps on the classpath...rather than the reverse in the profile case. Which is fine if you dont remove the dependencyManagement entries ensuring the desired versions are chosen.